### ISSUE TITLE
Unable to use build_event_stream.proto as external dependency

### ISSUE THREAD

**Issue by JSGette** [bazelbuild/bazel#16335]
### Description of the bug:

Instead of putting [build_event_stream.proto](https://github.com/bazelbuild/bazel/blob/master/src/main/java/com/google/devtools/build/lib/buildeventstream/proto/build_event_stream.proto) directly into my project to implement build event service I want to consume all required proto files by putting [bazel](https://github.com/bazelbuild/bazel) as external dependency in WORKSPACE file of my project:

```
http_archive(
   name = "bazel",
    build_file = "bazel.BUILD",
    sha256 = "149d475bd48cee245a3365bb523b6a702101800025dbf9aba217d059f3b6ef35",
    urls = ["https://github.com/bazelbuild/bazel/archive/refs/tags/5.3.1.tar.gz"],
    strip_prefix = "bazel-5.3.1",
)
```

Content of external/bazel.BUILD:
```
package(default_visibility = ["//visibility:public"])
load("@rules_java//java:defs.bzl", "java_proto_library")
load("@rules_proto//proto:defs.bzl", "proto_library")

java_proto_library(
    name = "build_event_stream_java_proto",
    deps = ["build_event_stream_proto"],
)

proto_library(
    name = "command_line_proto",
    srcs = ["//src/main/protobuf/command_line.proto"],
    deps = [":option_filters_proto"],
)

proto_library(
    name = "option_filters_proto",
    srcs = ["//src/main/protobuf/option_filters.proto"],
)

proto_library(
    name = "failure_details_proto",
    srcs = ["//src/main/protobuf/failure_details.proto"],
    deps = ["@com_google_protobuf//:descriptor_proto"],
)

proto_library(
    name = "invocation_policy_proto",
    srcs = ["//src/main/protobuf/invocation_policy.proto"],
)

proto_library(
    name = "build_event_stream_proto",
    srcs = ["src/main/java/com/google/devtools/build/lib/buildeventstream/proto/build_event_stream.proto"],
    deps = [
        ":command_line_proto",
        ":failure_details_proto",
        ":invocation_policy_proto",
        "@com_google_protobuf//:duration_proto",
        "@com_google_protobuf//:timestamp_proto",
    ],
)
```

This doesn't work as in [BUILD](https://github.com/bazelbuild/bazel/blob/master/src/main/java/com/google/devtools/build/lib/buildeventstream/proto/BUILD) default visibility is defined to subpackages of src package within bazel. 
```
package(default_visibility = ["//src:__subpackages__"])
```
Hence, it isn't possible to access proto-files directly without "patching" cloned repository and removing BUILD file.

Error:
```
BUILD.bazel:32:14: 
Label '@bazel//:src/main/java/com/google/devtools/build/lib/buildeventstream/proto/build_event_stream.proto' 
is invalid because '@bazel//src/main/java/com/google/devtools/build/lib/buildeventstream/proto' is a subpackage;
perhaps you meant to put the colon here: '@bazel//src/main/java/com/google/devtools/build/lib/buildeventstream/proto:build_event_stream.proto'?
```

### Which operating system are you running Bazel on?

WSL2 (Ubuntu)

### What is the output of `bazel info release`?

release 5.3.1

**Comment by sgowroji** [comment]
Adding the similar bug https://github.com/bazelbuild/bazel/issues/15874. Not the dupe.

**Comment by larsrc-google** [comment]
Seems very similar to bug #15874, so assigning to its team.

**Comment by haxorz** [comment]
Indeed #15874 sounds similar, thanks @sgowroji for noting the connection!

@JSGette, why do you want to consume all the required proto files? Would instead consuming an appropriate `x_proto_library` target work for your purposes? If so, what's your `x`? If not, what are you doing and do you really need visibility of the `.proto` file targets themselves?

**Comment by haxorz** [comment]
@michaeledgar fyi

**Comment by brentleyjones** [comment]
I also need this to be visible. Mine is `swift_proto_library`. The proto itself doesn't need to be visible, but the `proto_library`s do, otherwise I have to copy over the files and do this: https://github.com/buildbuddy-io/rules_xcodeproj/blob/62316bcfcc5ed56a83cf0f4771d57cf16a8e222a/tools/bep_parser/bep/BUILD#L7-L43

**Comment by JSGette** [comment]
@haxorz I wanted to consume this target:
`@bazel//src/main/java/com/google/devtools/build/lib/buildeventstream/proto:build_event_stream.proto`

**Comment by cameron-martin** [comment]
Related to https://github.com/bazelbuild/bazel/issues/3684

**Comment by Zemnmez** [comment]
I had this same issue today trying to make something that would consume BEP and make the appropriate output for github actions.

**Comment by michaeledgar** [comment]
This looks easy to correct. I'll check around for some historical context, but AFAICT this seems like a totally fine change to make: https://bazel-review.googlesource.com/c/bazel/+/238093

I'm not terribly persuaded that the BEP protobufs need to live in a different repository, though perhaps they belong with the other protobufs.

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 7.2.0 RC1](https://github.com/bazelbuild/bazel/releases/tag/7.2.0rc1). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting USE_BAZEL_VERSION=7.2.0rc1. Thanks!
